components/tidb_query/src/codec/row/v2/row_slice.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![allow(dead_code)]
+use crate::codec::{Error, Result};
+use codec::prelude::*;
+use num_traits::PrimInt;
+use tikv_util::codec::read_slice;
+
+#[derive(Debug)]
+struct RowSlice<'a> {
```

Comment:
Prefer to use an enum to define the `RowSlice`, which can get benefit from the compiler and the compiler will complain if we forget to check the `is_big` variable.

Reply:
Furthermore, define as enum will make the struct have a smaller size.